src/electron/device-connect-view/components/device-connect-port-entry.tsx

```diff
@@ -41,38 +42,35 @@ export class DeviceConnectPortEntry extends React.Component<DeviceConnectPortEnt
                     maskChar=""
                     mask="99999"
                 />
-                <Button
-                    primary={this.props.needsValidation}
-                    disabled={this.state.isValidateButtonDisabled}
-                    className="button-validate-port"
-                    onClick={this.onValidateClick}
-                >
-                    Validate port number
-                </Button>
+                {this.renderValidationPortButton()}
             </div>
         );
     }
 
-    private onPortTextChanged = (event: React.FormEvent<HTMLInputElement | HTMLTextAreaElement>, newValue?: string) => {
-        this.setState({ isValidateButtonDisabled: !newValue || newValue === '', port: newValue });
-        this.props.updateStateCallback(DeviceConnectState.Default);
```

Comment:
We lost this state change with the refactor:
1. Connect to device successfully
2. Edit port
3. Expected state: Expected state: start testing disabled, validate port button primary (the default state)
    Actual state: start testing enabled, validate port button not primary (the connected state)

It looks like the reason is because no message is sent back to the store as a part of the new `onPortTextChanged`, so no state update comes back through the props.

Reply:
we shouldn't be changing state when the text is being changed rite? may be thats the reason. we update state only when the user clicks validate?

Reply:
During the feature to implement this screen, the intentional design was that a change in port number after a validation has successfully occurred should return the state to pre-validation, since the new port number has not been validated.

Reply:
Good catch @jalkire I'll update the logic to account for this

Reply:
Updated!